Objects/genobject.c

```diff
@@ -405,22 +407,24 @@ _gen_throw(PyGenObject *gen, int close_on_genexit,
                We have to allow some awaits to work it through, hence the
                `close_on_genexit` parameter here.
             */
-            gen->gi_running = 1;
+            PyFrameState state = gen->gi_frame->f_state;
+            gen->gi_frame->f_state = FRAME_EXECUTING;
             err = gen_close_iter(yf);
-            gen->gi_running = 0;
+            gen->gi_frame->f_state = state;
```

Comment:
If the previous state was not FRAME_SUSPENDED, what would that mean, and should the generator really return to that state?  (applies several places)  I had sort of thought that the only backwards transitions were FRAME_EXECUTING to FRAME_SUSPENDED, and FRAME_CLEARED to FRAME_CREATED when the memory was reallocated to a different frame.

Reply:
I also think that the state must be `FRAME_SUSPENDED`, but if were some other state were possible this code would still be correct. So I'm inclined to leave it as is.